DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentProducer/IncidentProducer.cs

```diff
@@ -0,0 +1,215 @@
+                    output.properties.title += ". Object: " + prop.value;
+                    TextWriter(id, (string)prop.key, (string)prop.value, log);
+                    break;
+                case 3:
+                    output.properties.title += ". Source: " + prop.value;
+                    break;
+                }
+                i++;
+                if (i == 13) break;
+            }
+
+            lock (queueLock)
+            {
+                outputQueueItem.Add(JsonConvert.SerializeObject(output));
+            }
+        }
+
+        private static void WriteData(string storageConnectionString, string containerName, string path, string data)
+        {
+            if (CloudStorageAccount.TryParse(storageConnectionString, out var storageAccount))
+            {
+                // Create the CloudBlobClient that represents the Blob storage endpoint for the storage account.
+                var cloudBlobClient = storageAccount.CreateCloudBlobClient();
+                var container = cloudBlobClient.GetContainerReference(containerName);
+                var blobRef = container.GetBlockBlobReference(path);
+                var dataAsBytes = Encoding.UTF8.GetBytes(data);
+                blobRef.UploadFromByteArrayAsync(dataAsBytes, 0, dataAsBytes.Length).Wait();
+            }
+            else
+            {
+                throw new InvalidProgramException("Invalid format string");
+            }
+        }
+
+        private static void TextWriter(string incidentID, string param, string value, ILogger log)
+        {
+            try
+            {
+                const string container = "extra-parameters";
+                string Blob = incidentID + "\\" + param; // ID unique to the incident
+                var blobStorageConnectionString = Environment.GetEnvironmentVariable("BlobStorageConnectionString");
+                var blobStorageKeys = Environment.GetEnvironmentVariable("BlobStorageAccountKeys");
+                WriteData(blobStorageConnectionString, container, Blob, value);
+            }
+            catch (Exception ex)
+            {
+                log.LogError("Exception --> " + ex.Message);
+            }
+        }
+
+        [FunctionName("IncidentProducer")]
+        public static async Task RunAsync(
+#if DEBUG
+            [TimerTrigger("*/30 * * * * *")] TimerInfo myTimer,
+#else
+            [TimerTrigger("* */5 * * * *")]TimerInfo myTimer,
+#endif
+            [Queue("%CohesityQueueName%"), StorageAccount("AzureWebJobsStorage")] ICollector<string> outputQueueItem,
+            ILogger log)
+        {
+            log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+            long startDateUsecs = 0;
+
+            try
+            {
+                var db = Connection.GetDatabase();
+                string apiKey = Environment.GetEnvironmentVariable("apiKey");
+                string redisKey = Environment.GetEnvironmentVariable("workspace") + apiKey;
```

Comment:
With introducing the storage blob we don't really need storing timestamps in Redis anymore. We can re-use the same container and create there a separate blob with the name equal to Environment.GetEnvironmentVariable("workspace") + apiKey.

It'll reduce monthly bill for the customer.